mock/py/mockbuild/util.py

```diff
@@ -1202,6 +1202,11 @@ def setup_default_config_opts(unprivUid, version, pkgpythondir):
     config_opts['rpmbuild_command'] = '/usr/bin/rpmbuild'
     config_opts['dnf_disable_plugins'] = ['local', 'spacewalk']
 
+    # rpkg preprocessing configuration options
+    config_opts['rpkg_preprocessing'] = False
+    config_opts['rpkg_preprocessing_requires'] = ['preproc-rpmspec']
+    config_opts['rpkg_preprocessing_cmd'] = '/usr/bin/preproc-rpmspec %(source_spec)s --output %(target_spec)s'
```

Comment:
Instead of %() - can you use the built-in jinja2 expansion

Reply:
Hello, if I understand correctly, the jinja2 expansion can be only used to reference other configuration values. For example, config option `resultdir` uses `{{basedir}}` and `basedir` is defined earlier in `site-defaults.cfg`. But in this case, the values `source_spec` and `target_spec` cannot be completely preconfigured because they depend on name of the spec file being built into sprm which varies package by package. Am I understanding this correctly?

Reply:
Yes.  Similar problem has arisen in sign plugin, so we went with #497.
You can use `{{ rpkg_preprocessing.spec }}`.  But it is less nice, especially
considering that we should move this into mock plugin IMO and that would
add yet another level in the dict ... I'm not completely sure about the plugin,
yet.

Reply:
Alright, I thought the `{{ }}` braces can only be used to reference another configuration variable but they cannot be used to reference a variable, which is known only later during mock processing (when configuration has already been parsed). And when I looked into mock code, it didn't seem to me that you have some kind of special magic for it :). (Maybe I overlooked it).